qcodes/dataset/sqlite/database.py

```diff
@@ -150,14 +150,14 @@ def connect(name: str, debug: bool = False,
 
     # Make sure numpy ints and floats types are inserted properly
     for numpy_int in [
-        np.int, np.int8, np.int16, np.int32, np.int64,
+        np.int_, np.int8, np.int16, np.int32, np.int64,
```

Comment:
don't we need to add `int` similar to floats below? also, do i understand correctly that the current behavior (despite that names) is that python's builtin types are ALSO going through these adapters because np.int and np.float include int and float?

Reply:
We actually need the regular (non numpy) float type to go through the adapter to support correct handling of NaN. There is no such requirement for int.